snapcraft/tests/test_commands_upload.py

```diff
@@ -135,3 +135,41 @@ def test_just_upload_if_snap_file_exists(self):
             'snap-test_1.0_amd64.snap',
             'snap-test',
             config=self.mock_load_config.return_value)
+
+    def test_upload_nonexisting_snap_must_raise_exception(self):
+        with self.assertRaises(FileNotFoundError) as raised:
+            upload.main(['unexisting.snap'])
+
+        self.assertEqual('unexisting.snap', str(raised.exception))
+
+    def test_upload_existing_snap(self):
+        open('test.snap', 'w').close()
+
+        meta_path = os.path.join(os.getcwd(), 'squashfs-root', 'meta')
+        os.makedirs(meta_path)
+        with open(os.path.join(meta_path, 'snap.yaml'), 'w') as yaml_file:
+            yaml_file.write('name: snaptestname\n')
+
+        temp_dir = upload.tempfile.TemporaryDirectory
+
+        def create_patch(method, return_value):
+            patcher = mock.patch.object(temp_dir, method)
+            mock_method = patcher.start()
+            mock_method.return_value = return_value
+            self.addCleanup(patcher.stop)
+
+        create_patch('__enter__', os.getcwd())
```

Comment:
Adding `create_patch('_cleanup', None)` before the first `create_patch` call prevents the `ResourceWarning: Implicitly cleaning up <TemporaryDirectory...` warning message